dotnet/src/Connectors/Connectors.AzureAIInference/Core/ChatClientCore.cs

```diff
@@ -0,0 +1,468 @@
+            FrequencyPenalty = executionSettings.FrequencyPenalty,
+            PresencePenalty = executionSettings.PresencePenalty,
+            Model = modelId,
+            Seed = executionSettings.Seed,
+        };
+
+        switch (executionSettings.ResponseFormat)
+        {
+            case ChatCompletionsResponseFormat formatObject:
+                // If the response format is an Azure SDK ChatCompletionsResponseFormat, just pass it along.
+                options.ResponseFormat = formatObject;
+                break;
+
+            case string formatString:
+                // If the response format is a string, map the ones we know about, and ignore the rest.
+                switch (formatString)
+                {
+                    case "json_object":
+                        options.ResponseFormat = new ChatCompletionsResponseFormatJSON();
+                        break;
+
+                    case "text":
+                        options.ResponseFormat = new ChatCompletionsResponseFormatText();
+                        break;
+                }
+                break;
+
+            case JsonElement formatElement:
+                // This is a workaround for a type mismatch when deserializing a JSON into an object? type property.
+                // Handling only string formatElement.
+                if (formatElement.ValueKind == JsonValueKind.String)
+                {
+                    string formatString = formatElement.GetString() ?? "";
+                    switch (formatString)
+                    {
+                        case "json_object":
+                            options.ResponseFormat = new ChatCompletionsResponseFormatJSON();
+                            break;
+
+                        case "text":
+                            options.ResponseFormat = new ChatCompletionsResponseFormatText();
+                            break;
+                    }
+                }
+                break;
+        }
+
+        if (executionSettings.StopSequences is { Count: > 0 })
+        {
+            foreach (var s in executionSettings.StopSequences)
+            {
+                options.StopSequences.Add(s);
+            }
+        }
+
+        foreach (var message in chatHistory)
+        {
+            options.Messages.AddRange(GetRequestMessages(message /*, executionSettings.ToolCallBehavior */));
+        }
+
+        return options;
+    }
+
+    private static List<ChatRequestMessage> GetRequestMessages(ChatMessageContent message)
+    {
+        if (message.Role == AuthorRole.System)
+        {
+            return [new ChatRequestSystemMessage(message.Content)];
+        }
+
+        if (message.Role == AuthorRole.User)
+        {
+            if (message.Items is { Count: 1 } && message.Items.FirstOrDefault() is TextContent textContent)
+            {
+                return [new ChatRequestUserMessage(textContent.Text) /*{ Name = message.AuthorName }*/ ];
```

Comment:
There are multiple comments like this in the code, should we create an issue to handle them later?

Reply:
Actually I leave it there to add when this issue is fixed.

- https://github.com/Azure/azure-sdk-for-net/issues/45415